frontend/public/reducers/ui.ts

```diff
@@ -59,6 +59,26 @@ const silenceFiringAlerts = (firingAlerts, silences) => {
   });
 };
 
+const silenceFiringRules = (firingRules, silences) => {
```

Comment:
Looks like we can make this much simper by removing `silenceFiringRules` and leaving `MonitoringSetRules` unchanged. Then modify `silenceFiringAlerts` instead.

`silenceFiringAlerts` is already being run when the rules are imported, and already does much of the work to determine which alerts are silenced.

Modify `silenceFiringAlerts ` by adding the following (or something similar) inside the `if (a.silencedBy.length) { ... }` block.
```
if (!_.isEmpty(a.rule.alerts) && _.every(a.rule.alerts, isSilenced)) {
  a.rule.state = RuleStates.Silenced;
  a.rule.silencedBy = _.filter(
    silences?.data,
    (s) => s.status.state === SilenceStates.Active && _.some(a.rule.alerts, isSilenced),
  );
}
```
(briefly tested, but not properly)